platform/spire/src/setup.py

```diff
@@ -141,6 +145,31 @@ def setup_keyserver(ops: Operations) -> None:
         ops.ssh("enable keyserver on @HOST", node, "systemctl", "enable", "keyserver.service")
         ops.ssh("start keyserver on @HOST", node, "systemctl", "restart", "keyserver.service")
 
+def redeploy_keyserver(ops: Operations) -> None:
+    config = configuration.get_config()
+    for node in config.nodes:
+        if node.kind != "supervisor":
+            continue
+        # delete the existing configs
+        ops.ssh_rm("delete existing cluster config from @HOST", node, STATICS_DIR + "/cluster.conf")
+        ops.ssh_rm("delete existing keyserver config from @HOST", node, CONFIG_DIR + "/setup.yaml")
+        # redeploy new config
+        ops.ssh_upload_bytes("reupload cluster config to @HOST", node,
+            configuration.get_cluster_conf().encode(), STATICS_DIR + "/cluster.conf")
+        ops.ssh_upload_path("upload cluster setup to @HOST", node,
+                            configuration.Config.get_setup_path(), CONFIG_DIR + "/setup.yaml")
+        # restart the keyserver
+        ops.ssh("restart keyserver on @HOST", node, "systemctl", "restart", "keyserver.service")
+
+def redeploy_keyclients(ops: Operations) -> None:
+    config = configuration.get_config()
+    for node in config.nodes:
+        # delete existing cluster configuration from non-supervisor nodes
+        if node.kind != "supervisor":
+            ops.ssh_rm("delete existing cluster config from @HOST", node, CONFIG_DIR + "/cluster.conf")
```

Comment:
Can you explain in comments why we don't delete the configs from the supervisor?

Reply:
I have changed the line `# delete existing cluster configuration from non-supervisor nodes` to 
```python
        # we do not delete the config on the supervisor because the supervisor
        # client config and keyserver config are in the same location
```

To reflect why we don't delete the config from the supervisor. As you mentioned earlier, the keyserver and keyclient probably should be using the same config directory, maybe this is something we should change here too? (Also i might rewrite this comment again, perhaps the language
```python
        # do not delete the cluster.conf on the supervisor because the supervisor
        # keyclient and keyserver use the same cluster.conf
        # TODO(alvareza): modify keyclient and keyserver to use different config directories
```
would work?

Reply:
I'd file an issue, rather than adding a TODO to the codebase, especially because it's not really a TODO to change something in this code exactly.

Reply:
I have filed a new issue to track this issue, see #475. Furthermore, I added the comment (minus the todo) in 2f980ef4e2116cb750b6cb5e765bed381c4eab68